### PR TITLE
Support for the online installation medium (jsc#SLE-7214)

### PR DESCRIPTION
**Pull request by lslezak**
- Adjust the product selection in the welcome screen for the online medium installation
  - The products are read from the `control.xml` file
  - Updated selecting the product
- https://jira.suse.com/browse/SLE-7214
- Tested manually
- 4.2.14

### CODE REVIEW

File: src/lib/installation/clients/inst_complex_welcome.rb [hunk 1]
```diff
@@ -19,11 +19,15 @@
 # current contact information at www.novell.com.
 # ------------------------------------------------------------------------------
 
-require "yaml"
 require "fileutils"
+require "pp"
```
Comment (jreidinger) [inline src/lib/installation/clients/inst_complex_welcome.rb]: why it is needed?

File: src/lib/installation/widgets/product_selector.rb [hunk 2]
```diff
@@ -79,6 +86,10 @@ def registered?
       rescue LoadError
         false
       end
+
+      def product_id(prod)
```
Comment (jreidinger) [inline src/lib/installation/widgets/product_selector.rb]: hmm, can we extend prod to provide this method? For me it does not belong here, but to prod object itself.

File: src/lib/installation/clients/inst_complex_welcome.rb (hunk 1 above)
Comment (lslezak) [reply src/lib/installation/clients/inst_complex_welcome.rb]: To better format the logged data:
```ruby
log.info "Found base products in the control.xml: #{@available_base_products.pretty_inspect}"
```

File: src/lib/installation/widgets/product_selector.rb (hunk 2 above)
Comment (lslezak) [reply src/lib/installation/widgets/product_selector.rb]: It's actually a widget ID, it not related to the products at all, it's used for identifying the RadioButton widgets. So I'd leave it here. I just renamed it to `widget_id` to make it clear.

Comment (lslezak) [reply src/lib/installation/widgets/product_selector.rb]: ah, with "widget_id" it fails, that's already used by CWM, renaming to "item_id"...

**Pull request merged**